examples/spec/integration/serial_hello_world_workflow_spec.rb

```diff
@@ -3,6 +3,37 @@
 describe SerialHelloWorldWorkflow do
   subject { described_class }
 
+  it 'works' do
+    Temporal.configure do |config|
+      config.host = 'temporal'
+      config.port = 7233
+      config.namespace = 'ruby-samples'
+      config.task_queue = 'general'
+    end
+
+    workflow_id = SecureRandom.uuid
+    run_id = Temporal.start_workflow(
+      SerialHelloWorldWorkflow,
+      'Alice',
+      'Bob',
+      'John',
+      options: { workflow_id: workflow_id }
+    )
+
+    client = Temporal.send(:client)
+
+    result = client.get_workflow_execution_history(
+      namespace: Temporal.configuration.namespace,
+      workflow_id: workflow_id,
+      run_id: run_id,
+      next_page_token: nil,
+      wait_for_new_event: true,
```

Comment:
I imagine even with `wait_for_new_event` you
(a) need to run this command first to avoid the case where the workflow finishes first.  (unless `wait_new_event` returns immediately if the the workflow has already hit a terminal state?)
(b) still need a polling loop until it gets to a terminal state in case there are intermediary events.

Reply:
Great points:

(a) I don't think you can run this first since you won't have a run_id at that time. I'll test it, but I'm pretty sure it'll return new events since the last call
(b) `event_type: :close` will filter out any intermediary events